grove/tomography/tomography.py

```diff
@@ -0,0 +1,727 @@
+            c_jk_m_elms[(j + k * readout_povm.pi_basis.dim, m)] = val.real
+
+    # create sparse matrix from COO-format (see scipy.sparse docs)
+    _keys, _values = izip(*c_jk_m_elms.items())
+    _rows, _cols = izip(*_keys)
+    c_jk_m = coo_matrix((list(_values), (list(_rows), list(_cols))),
+                        shape=(readout_povm.pi_basis.dim * len(channel_ops),
+                               pauli_basis.dim)).tocsr()
+    return c_jk_m
+
+
+def _prepare_b_jkl_mn(readout_povm, pauli_basis, pre_channel_ops, post_channel_ops, rho0):
+    """
+    Prepare the coefficient matrix for process tomography. This function uses sparse matrices
+    for much greater efficiency. The coefficient matrix is defined as:
+
+    .. math::
+
+            B_{(jkl)(mn)}=\sum_{r,q}\pi_{jr}(\mathcal{R}_{k})_{rm} (\mathcal{R}_{l})_{nq} (\rho_0)_q
+
+    where :math:`\mathcal{R}_{k}` is the transfer matrix of the quantum map corresponding to the
+    k-th pre-measurement channel, while :math:`\mathcal{R}_{l}` is the transfer matrix of the l-th
+    state preparation process. We also require the overlap
+    between the (generalized) Pauli basis ops and the projection operators
+    :math:`\pi_{jl}:=\sbraket{\Pi_j}{P_l} = \tr{\Pi_j P_l}`.
+
+    See the grove documentation on tomography for detailed information.
+
+    :param DiagonalPOVM readout_povm: The POVM corresponding to the readout plus classifier.
+    :param OperatorBasis pauli_basis: The (generalized) Pauli basis employed in the estimation.
+    :param list pre_channel_ops: The state preparation channel operators as `qutip.Qobj`
+    :param list post_channel_ops: The pre-measurement (post circuit) channel operators as `qutip.Qobj`
+    :param qutip.Qobj rho0: The initial state as a density matrix.
+    :return: The coefficient matrix necessary to set up the binomial state tomography problem.
+    :rtype: scipy.sparse.csr_matrix
+    """
+    c_jk_m = _prepare_c_jk_m(readout_povm, pauli_basis, post_channel_ops)
+    pre_channel_transfer_matrices = [pauli_basis.transfer_matrix(qt.to_super(ek))
+                                     for ek in pre_channel_ops]
+    rho0_q = pauli_basis.project_op(rho0)
+
+    # These next lines hide some very serious (sparse-)matrix index magic,
+    # basically we exploit the same index math as in `qutip.sprepost()`
+    # i.e., if a matrix X is linearly mapped `X -> A.dot(X).dot(B)`
+    # then this can be rewritten as
+    #           `np.kron(B.T, A).dot(X.T.ravel()).reshape((B.shape[1], A.shape[0])).T`
+    # The extra matrix transpose operations are necessary because numpy by default
+    # uses row-major storage, whereas these operations are conventionally defined for column-major
+    # storage.
+    d_ln = spvstack([(rlnq * rho0_q).T for rlnq in pre_channel_transfer_matrices]).tocoo()
+    b_jkl_mn = spkron(d_ln, c_jk_m).real
+    return b_jkl_mn
+
+
+class StateTomography(object):
```

Comment:
can we implement a `TomographyBase` that shows what are commonalities between the classes and help the IDE identify method overloading?